src/main/scala/org/broadinstitute/dsde/workbench/sam/azure/AzureService.scala

```diff
@@ -84,64 +129,103 @@ class AzureService(crlService: CrlService, directoryDAO: DirectoryDAO) {
+    // if not there, for backwards compatibility, get the billing profile id from a tag on the Azure resource
+    OptionT(getBillingProfileIdFromSamDb(request, samRequestContext))
+      .orElseF(getBillingProfileIdFromAzureTag(request, samRequestContext))
+      .value
+
+  private def getBillingProfileIdFromSamDb(request: GetOrCreatePetManagedIdentityRequest, samRequestContext: SamRequestContext): IO[Option[BillingProfileId]] =
     for {
-      resourceManager <- crlService.buildResourceManager(request.tenantId, request.subscriptionId)
-      mrg <- IO(resourceManager.resourceGroups().getByName(request.managedResourceGroupName.value)).attempt
-    } yield mrg.toOption.flatMap(getBillingProfileFromTag)
+      maybeMrg <- azureManagedResourceGroupDAO.getManagedResourceGroupByCoordinates(request.toManagedResourceGroupCoordinates, samRequestContext)
+    } yield maybeMrg.map(_.billingProfileId)
 
-  /** Validates a managed resource group. This is only called at managed identity creation time. Algorithm:
-    *   1. Resolve the MRG in Azure 2. Get the managed app id from the MRG 3. Resolve the managed app as the publisher 4. Get the managed app "plan" id 4.
-    *      Validate the plan id matches the configured value
+  private def getBillingProfileIdFromAzureTag(
+      request: GetOrCreatePetManagedIdentityRequest,
+      samRequestContext: SamRequestContext
+  ): IO[Option[BillingProfileId]] = traceIOWithContext("getBillingProfileIdFromAzureTag", samRequestContext) { _ =>
+    for {
+      mrg <- validateManagedResourceGroup(request.toManagedResourceGroupCoordinates, samRequestContext, false)
+    } yield getBillingProfileFromTag(mrg)
+  }
+
+  /** Validates a managed resource group. Algorithm:
+    *   1. Resolve the MRG in Azure 2. Get the managed app id from the MRG 3. Resolve the managed app 4. Get the managed app "plan" name and publisher 5.
+    *      Validate the plan name and publisher matches a configured value 6. Validate that the caller is on the list of authorized users for the app
+    */
+  private def validateManagedResourceGroup(
+      mrgCoords: ManagedResourceGroupCoordinates,
+      samRequestContext: SamRequestContext,
+      validateUser: Boolean = true
+  ): IO[ResourceGroup] =
+    traceIOWithContext("validateManagedResourceGroup", samRequestContext) { _ =>
+      for {
+        resourceManager <- crlService.buildResourceManager(mrgCoords.tenantId, mrgCoords.subscriptionId)
+        mrg <- lookupMrg(mrgCoords, resourceManager)
+        appManager <- crlService.buildApplicationManager(mrgCoords.tenantId, mrgCoords.subscriptionId)
+        appsInSubscription <- IO(appManager.applications().list().asScala)
+        managedApp <- IO.fromOption(appsInSubscription.find(_.managedResourceGroupId() == mrg.id()))(managedAppValidationFailure)
+        plan <- validatePlan(managedApp, crlService.getManagedAppPlans)
+        _ <- if (validateUser) validateAuthorizedAppUser(managedApp, plan, samRequestContext) else IO.unit
```

Comment:
It probably doesn't matter if BPM is the only thing calling this API at billing-profile creation time -- but do we want to check `authorizedTerraUser` in Sam, or just rely on the `set_managed_resource_group` action?

Reply:
as it stands, anyone can make a spend-profile in sam then call this new api so I think this check is necessary